pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -19,12 +19,175 @@ package kuberuntime
 import (
 	"fmt"
 	"io"
+	"os"
+	"path"
 
+	"github.com/golang/glog"
 	"k8s.io/kubernetes/pkg/api"
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/types"
 	"k8s.io/kubernetes/pkg/util/term"
 )
 
+// getContainerLogsPath gets log path for container.
+func getContainerLogsPath(containerName string, podUID types.UID) string {
+	return path.Join(podLogsRootDirectory, string(podUID), fmt.Sprintf("%s.log", containerName))
+}
+
+// generateContainerConfig generates container config for kubelet runtime api.
+func (m *kubeGenericRuntimeManager) generateContainerConfig(container *api.Container, pod *api.Pod, restartCount int, podIP string) (*runtimeApi.ContainerConfig, error) {
+	opts, err := m.runtimeHelper.GenerateRunContainerOptions(pod, container, podIP)
+	if err != nil {
+		return nil, err
+	}
+
+	_, containerName, cid := buildContainerName(pod, container)
```

Comment:
That's actually exactly what labels are for, right? If you trust the runtime to persist labels as well as it persists names, then agreed we don't need this crazy name.

If the runtime chooses to do this translation internally so it can better recover or identify or whatever, that's fine too.